src/AMSlib/wf/basedb.hpp

```diff
@@ -792,6 +813,46 @@ struct AMSMsgHeader {
 
     return AMSMsgHeader::size();
   }
+
+  /**
+   * @brief Return a valid header based on a pre-existing data buffer
+   * @param[in] data_blob The buffer to fill
+   * @return An AMSMsgHeader with the correct attributes
+   */
+  static AMSMsgHeader decode(uint8_t* data_blob)
+  {
+    size_t current_offset = 0;
+    // Header size (should be 1 bytes)
+    uint8_t new_hsize = data_blob[current_offset];
+    CWARNING(AMSMsgHeader,
+             new_hsize != AMSMsgHeader::size(),
+             "buffer is likely not a valid AMSMessage (%d / %d)",
+             new_hsize,
+             current_offset)
+
+    current_offset += sizeof(uint8_t);
+    // Data type (should be 1 bytes)
+    uint8_t new_dtype = data_blob[current_offset];
+    current_offset += sizeof(uint8_t);
+    // MPI rank (should be 2 bytes)
+    uint16_t new_mpirank;
+    std::memcpy(&new_mpirank, data_blob + current_offset, sizeof(uint16_t));
```

Comment:
You can avoid memcpy's here (that can be expensive).
`new_mpirank = (static_cast<uint16_t*>(data_blob + current_offset))[0];`

You may need to use `reinterpret_cast`.

Reply:
Fixed, thanks.